CHANGELOG.md

```diff
@@ -21,6 +21,7 @@
   error serialization is enabled in Tarantool (gh-376).
 - Added `assert_items_exclude`.
 - Strip useless `...` lines from error trace.
+- Make `assert_error_*` ignore errors raised by assertion failures.
```

Comment:
Nit: I would say 're-raise' instead of 'ignore', because the latter can mean 'suppress'.

Reply:
Dropped this patch. Now I wrap exec error with a special error class and unwrap only those errors in `t.assert_error_*`, as you initially suggested. Please take another look.